telemetry/telemetry.proto

```diff
@@ -4,6 +4,57 @@ import 'nanopb.proto';
 
 option java_outer_classname="TelemetryProto";
 
-message Telemetry {
-  optional int32 timestamp_ms = 1;
+//Should be sent only after device resets
+message NameIDMessage { //Send the device ID with its corresponding name
+	optional uint32 id = 1;
+	optional string name = 2 [(nanopb).max_size = 32];
+}
+
+message StatisticalAggregate {
+  optional uint32 numSamples = 1;
+  optional uint32 min = 2;
+  optional uint32 max = 3;
+  optional uint32 mean = 4;
+  optional uint32 stdev = 5;
+}
+
+message GeneralReading { //this might want a more specific name?
+	enum READING_TYPE {
+		VOLTAGE = 1;
+		CURRENT = 2;
+		TEMPERATURE = 3;
+	}
+	optional READING_TYPE type = 1;
```

Comment:
Yeah, I think the resolution / accuracy / sensor limits should be statically defined in a config file. It doesn't make sense to calculate those on-the-fly, since they're completely defined by the hardware.

Units: I would put them with the other metadata, and have the data be consistently defined in terms of those units. This is the best in terms of simplicity and precision (assuming your readings are linear with the measurement).
There may be a slight efficiency hit, as protobufs use [varint encoding](https://developers.google.com/protocol-buffers/docs/encoding) where larger numbers (can) result in larger wire sizes. I think every 7 bits of number costs 8 bits. I don't think it's worth optimizing for this right now though.